### ISSUE TITLE
Missing $debugHintsPath when sending email via command

### ISSUE THREAD

**Issue by cykirsch** [magento/magento2#10440]
### Preconditions

1. Magento 2.1

### Steps to reproduce

1. Create a console command that sends an email, where the email also loads in another block like `{{layout handle="sales_email_order_items" order=$order area="frontend"}}`

### Expected result

1. Email should send without error.

### Actual result

1. Email fails with error
```
main.CRITICAL: exception 'BadMethodCallException' with message 'Missing required argument $debugHintsPath of Magento\Developer\Model\TemplateEngine\Plugin\DebugHints.' in /.../.../magento/vendor/magento/framework/ObjectManager/Factory/Dynamic/Developer.php:45
```

### More info
A workaround was found and posted on these sites:
https://community.magento.com/t5/Programming-Questions/Email-template-with-block/m-p/50061#M1219
https://magento.stackexchange.com/questions/152089/magento2-command-line-sending-email-using-block-templates-error-missing-r

Unfortunately it requires invoking object manager directly, as it seems the object manager's configuration is what needs to be corrected. Since direct loading of the object manager is not best practice, if modifying this config is the only solution then a method should be provided to modify the config without loading object manager directly.

**Comment by magento-engcom-team** [comment]
@cykirsch, thank you for your report.
We've created internal ticket(s) MAGETWO-81170 to track progress on the issue.

**Comment by nemesis-back** [comment]
@okorshenko I will work on it #SQUASHTOBERFEST

**Comment by orlangur** [comment]
Implementation approach described in https://github.com/magento/magento2/pull/11852#pullrequestreview-132885912

**Comment by passtet** [comment]
#khcd2018

**Comment by magento-engcom-team** [comment]
@passtet thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by magento-engcom-team** [comment]
Hi @cykirsch. Thank you for your report. 
The issue has been fixed in magento/magento2#17984 by @passtet in 2.2-develop branch
Related commit(s):
 - [a804a3cd485bd4aecabac3aa4203d20c6a6c9a08](https://github.com/magento/magento2/commit/a804a3cd485bd4aecabac3aa4203d20c6a6c9a08)

The fix will be available with the upcoming 2.2.8 release.

**Comment by sidolov** [comment]
Hi @cykirsch. Thank you for your report. 
The issue has been fixed in magento/magento2#18991 by @p-bystritsky in 2.3-develop branch
Related commit(s):
 - [7c86822387cbee24186a1746b373b2bfb381bed8](https://github.com/magento/magento2/commit/7c86822387cbee24186a1746b373b2bfb381bed8)

The fix will be available with the upcoming 2.3.1 release.
